tools/generate-helpers/pg-table-bindings/store.go.tpl

```diff
@@ -528,8 +531,42 @@ func (s *storeImpl) Get(ctx context.Context, {{template "paramList" $pks}}) (*{{
     } else if !ok {
         return nil, false, nil
     }
+    {{- else if .Obj.IsDirectlyScoped }}
+    scopeChecker := sac.GlobalAccessScopeChecker(ctx)
+	scopeTree, err := scopeChecker.EffectiveAccessScope(permissions.ResourceWithAccess{
+		Resource: targetResource,
+		Access:   storage.Access_READ_ACCESS,
+	})
+	if err != nil {
+        return nil, false, err
+	}
+    {{- if .Obj.IsClusterScope }}
+    sacQueryFilter, err = sac.BuildClusterLevelSACQueryFilter(scopeTree)
+    {{- else}}
+    sacQueryFilter, err = sac.BuildClusterNamespaceLevelSACQueryFilter(scopeTree)
+    {{- end }}
+	if err != nil {
+        return nil, false, err
+	}
+    {{- end }}
+{{/* TODO(ROX-10624): Remove this condition after all PKs fields were search tagged (PR #1653) */}}
+{{- if eq (len $pks) 1 }}
+    q := search.ConjunctionQuery(
+    sacQueryFilter,
+    {{- range $idx, $pk := $pks}}
+        {{- if eq $pk.Name $singlePK.Name }}
+            search.NewQueryBuilder().AddDocIDs({{ $singlePK.ColumnName|lowerCamelCase }}).ProtoQuery(),
+        {{- else }}
+            search.NewQueryBuilder().AddExactMatches(search.FieldLabel("{{ $pk.Search.FieldName }}"), {{ $pk.ColumnName|lowerCamelCase }}).ProtoQuery(),
+        {{- end}}
```

Comment:
Why the fork? Can we have a unified approach use the search labels associated with the primary keys?

Reply:
We can't. Most of single PK resources don't have search tag for ID. There are some exceptions listed below
```
# git grep  'sql:\"pk' -- '*.proto' | grep search

proto/storage/cluster.proto:    string                    id                               = 1 [(gogoproto.moretags) = 'search:"Cluster ID,hidden,store" sql:"pk"'];
proto/storage/cve.proto:    string                    id                    = 1 [(gogoproto.moretags) = 'search:"CVE ID,hidden" sql:"pk,id"'];  // This field is composite of cve and operating system.
proto/storage/deployment.proto:    string id                                  = 1 [(gogoproto.moretags) = 'search:"Deployment ID,store,hidden" sql:"pk"'];
proto/storage/image.proto:    string                          id                          = 4 [(gogoproto.moretags) = 'search:"Image Sha,store,hidden" sql:"pk"'];
proto/storage/image_component.proto:    string                         id               = 1 [(gogoproto.moretags) = 'search:"Component ID,store,hidden" sql:"pk,id"']; // This field is composite id over name, version, and operating system.
proto/storage/namespace_metadata.proto:    string                    id            = 1 [(gogoproto.moretags) = 'search:"Namespace ID" sql:"pk"'];
proto/storage/policy.proto:    string                      id                      = 1 [(gogoproto.moretags) = 'search:"Policy ID,store,hidden" sql:"pk"'];
proto/storage/process_indicator.proto:    string id                                      = 1 [(gogoproto.moretags) = 'search:"Process ID,store,hidden"  sql:"pk"'];
proto/storage/rbac.proto:    string id                               = 1 [(gogoproto.moretags) = 'search:"Role ID,hidden" sql:"pk"'];
proto/storage/rbac.proto:    string id                               = 1 [(gogoproto.moretags) = 'search:"Role Binding ID,hidden" sql:"pk"'];
proto/storage/secret.proto:    string              id               = 1 [(gogoproto.moretags) = 'search:"Secret ID,store,hidden" sql:"pk"'];
proto/storage/test.proto:    string key                 = 1 [(gogoproto.moretags) = 'sql:"pk,index=hash" search:"Test Key"'];
proto/storage/test.proto:  string key1                  = 1 [(gogoproto.moretags) = 'sql:"pk,id" search:"Test Key"'];
proto/storage/test.proto:  string id  = 1 [(gogoproto.moretags) = 'search:"Test Grandparent ID" sql:"pk"'];
proto/storage/test.proto:   string id        = 1 [(gogoproto.moretags) = 'search:"Test Parent1 ID" sql:"pk"'];
proto/storage/test.proto:  string id = 1 [(gogoproto.moretags) = 'search:"Test Child1 ID" sql:"pk"'];
proto/storage/test.proto:  string id        = 1 [(gogoproto.moretags) = 'search:"Test Grandchild1 ID" sql:"pk"'];
proto/storage/test.proto:  string id = 1 [(gogoproto.moretags) = 'search:"Test GGrandchild1 ID" sql:"pk"'];
proto/storage/test.proto:  string id        = 1 [(gogoproto.moretags) = 'search:"Test G2Grandchild1 ID" sql:"pk"'];
proto/storage/test.proto:  string id = 1 [(gogoproto.moretags) = 'search:"Test G3Grandchild1 ID" sql:"pk"'];
proto/storage/test.proto:  string id        = 1 [(gogoproto.moretags) = 'search:"Test Parent2 ID" sql:"pk"'];
proto/storage/test.proto:  string id        = 1 [(gogoproto.moretags) = 'search:"Test Child2 ID" sql:"pk"'];
proto/storage/test.proto:  string id        = 1 [(gogoproto.moretags) = 'search:"Test Parent3 ID" sql:"pk"'];
```